docs/DeveloperGuide.md

```diff
@@ -1,34 +1,561 @@
+and consists of the timetable Command and CommandParser classes, as well as TimeTableManager and TimeTable themselves. 
+Extending from the abstract TimeTableCommand class are the TimeTableAddCommand, TimeTableDeleteCommand and TimeTableViewCommand classes.
+
+![Class diagram for TimeTable Family Classes](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/TimeTableClassDiagram.png?raw=true)
+
+**This is a good representation of how the other data classes (Task, Module) work as well.** The Command objects
+call the methods held in the Manager classes to perform work on the stored user data.
+
+Upon the first start up of ra.VI, `TimeTableManager.initialise()` will be run. This will no longer run again as long as the user 
+does not tamper with or delete the files in the created data folder.
+
+The TimeTable is created based on the user's initial input, with an appropriate number of LessonManagers.
+The point of entry for this feature will be at TimeTableCommandParser, which will decide which of the commands 
+to return through `parseTimeTableCommand()`. If the TimeTableCommand is returned and executed, the 
+TimeTableManager will carry out the associated commands, adding, deleting or viewing the Lessons in the timetable.
+
+## Feature explanation with sequence diagrams
+
+### Main loop sequence
+When ra.VI runs, there are 3 phases to its lifecycle.
+
+1. Initialisation
+2. Main command loop
+3. Saving and exiting
+
+The main sequence diagram can be broken into three parts representing each of these phases.
+
+![Sequence diagram 1 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram1.png?raw=true)
+ra.VI's `start()` method initialises all the classes used by the main class, like TextUi, InputOutputManager, and more.
+TimeTableManager may need user input to be initialised, and so it is placed in a validation loop.
+A welcome message is shown at the end of the initialisation phase.
+
+![Sequence diagram 2 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram2.png?raw=true)
+The main loop is contained in the 'runCommandLoopUntilExitCommand()' method.
+The main loop follows the following steps:
+1. Get the user command as a string.
+2. Parse the command using Parser. Parser will return a Command object.
+3. Execute the Command object with the `execute()` method. This will do the necessary work and return a CommandResult object.
+4. If any data was changed, StateManager will run `saveState()` to facilitate undo commands.
+5. Finally, use the CommandResult object to show the result of the command to the user using TextUi.
+
+Note that the Command and CommandResult objects are destroyed at the end of the method. Additionally, the Parser object
+is destroyed after each use.
+
+![Sequence diagram 3 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram3.png?raw=true)
```

Comment:
These diagrams are extremely complex and there is a lot of information. While a lot of effort has been put into annotating the information, I feel like it could be easier to comprehend if it was split into simpler subsections. 

For example, since you only want to talk about the `start()` command at the beginning, you could do a diagram on which functions are triggered by `start()`, then for 2nd one you can cover all functions run by `runCommandLoopUntilExitCommand()`. Focusing on just these functions alone (and not what invokes the functions) would help reduce the amount of stacking activation bars as well as give you a nice simple diagram for the 1st one with less redundant classes. If you still want to cover the entire execution of `main()` and `run()` you can add those in a separate diagram with no additional classes to cover the broad level scope of the program invocation and functions called.